tests/flow/test_evaluation_imperative.py

```diff
@@ -655,3 +655,45 @@ def test_evaluation_invalid_model_name_fixable(model_name):
 def test_evaluation_invalid_model_name_not_fixable(model_name):
     with pytest.raises(ValueError):
         weave.EvaluationLogger(model=model_name)
+
+
+def test_evaluation_logger_with_predefined_scorers(client, caplog):
+    """Test that EvaluationLogger can track predefined scorers and warn when using unlisted ones."""
+    import logging
+
+    # Create evaluation with predefined scorer names
+    ev = EvaluationLogger(
+        model="test_model",
+        dataset=[{"input": 1}],
+        scorers=["accuracy", "precision"],  # List of allowed scorer names
```

Comment:
I think `scorers` is an awkward name here, since there are no scorers with `EvaluationLogger`.

Two options discussed with @gtarpenning :
1. rename to `scorers` or `metrics`
2. instead of this awkward list of strings, let users specify a version number manually.

Option 2 is the most elegant and gives users the most control over versioning. The problem is that this version number would be out of sync with `version_index`. One way to get around that would be to call it `version_seed` or something, to signify that changing it bumps the version, but it's not great. It makes option 1 more compelling.

WDYT @andrewtruong ?

Reply:
I'm ok with the name `scorers`.

If this field is specified, it might be good to give a warning to the user if they end up logging a score with a name not in the provided list.